services/kingdom_setters.js

```diff
@@ -0,0 +1,68 @@
+const fs = require("fs");
+
+const getJson = () => {
+  const data = fs.readFileSync("./kingdom.json");
+  const json = JSON.parse(data);
+  return json;
+};
+
+const saveJson = json => {
+  fs.writeFileSync("./kingdom.json", JSON.stringify(json, null, 2));
+};
+
+const addKingdom = (kingdomName, kingName, queenName) => {
+  let json = getJson();
+
+  //check for existing object
+  if (json.kingdoms[kingdomName]) return;
+
+  json.kingdoms[kingdomName] = {};
+  json.kingdoms[kingdomName]["king"] = kingName;
+  json.kingdoms[kingdomName]["queen"] = queenName;
+  json.kingdoms[kingdomName]["castles"] = {};
+
+  saveJson(json);
```

Comment:
A lot of these functions have some structural duplication. At one level it's:

1. Get Json — always the same
2. Modify Json — different
3. Save Json — always the same

Try to think how you might write a method that allowed you to abstract this concept of a JSON transaction. I know this is the approach we took in the demo, so consider this a fun little challenge.